keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/diff/PartialDerivative.java

```diff
@@ -156,15 +177,34 @@ public static PartialDerivative matrixMultiplyAlongWrtDimensions(PartialDerivati
         return new PartialDerivative(result);
     }
 
-    public static DoubleTensor increaseRankByAppendingOnesToShape(DoubleTensor lowRankTensor, int desiredRank) {
-        return lowRankTensor.reshape(
-            TensorShape.shapeDesiredToRankByAppendingOnes(lowRankTensor.getShape(), desiredRank)
-        );
+    /**
+     * This is important for the case where the partial 'of' and the tensor are different ranks but are
+     * still broadcastable.
+     *
+     * @param tensor        the tensor to align along the of dimensions
+     * @param partialShape  the full shape of the partial in the format [of,wrt]
+     * @param partialOfRank the rank of the 'of' part of the partial
+     * @return a reshaped tensor with a shape of ones everywhere except the aligned of dimensions.
+     * E.g.
+     * partialShape = [2,3,4,5,6,7]
+     * partialOfRank = 3
+     * tensorShape = [3,4]
```

Comment:
nit: make these match the order of the params.

I find this method really confusing. It returns a shape of ones everywhere except where `tensorShape` does not equal `partialShape`? The example you've given seems a bit different from the `return` statement